### ISSUE TITLE
Prefetch value has no effect when using receive()

### ISSUE THREAD

**Issue by gabrielpallotta** [reactor/reactor-kafka#279]
When using the KafkaReceiver `.receive()`, the prefetch value passed as parameter (which has default value 1) seems to not influence the number of messages prefetched in memory. For example, setting it to the default value actually causes it to keep 256 batches in memory during processing.

Adding a prefetch parameter to the `.flatMapIterable()` in the `receive()` method solves this issue:
```java
  @Override
  public Flux<ReceiverRecord<K, V>> receive(Integer prefetch) {
    return withHandler(AckMode.MANUAL_ACK, (scheduler, handler) -> handler
        .receive()
        .publishOn(scheduler, preparePublishOnQueueSize(prefetch))
        .flatMapIterable(it -> it, preparePublishOnQueueSize(prefetch)) // configure prefetch as well
        .map(record -> new ReceiverRecord<>(
            record,
            handler.toCommittableOffset(record)
        )));
  }
```

Is this an expected behaviour? Or is there a correct way to configure this prefetch? The behaviour I am looking for is avoiding having lots of message batches in memory, ideally just the one that is currently being processed.

## Expected Behavior
With prefetch = 1, only one batch at a time would be kept in memory

## Actual Behavior
Even with low prefetch values, 256 batches are prefetched to memory

## Steps to Reproduce
Create a simple consumer with `KafkaReceiver.create(...).receive(1)`, and attach to a topic with a lot of lag. By enabling the debug logs on kafka reactor, we can see consuming from Kafka happens in spikes instead of having a constant rate (one batch at a time).

## Possible Solution
Adding a prefetch to the `.flatMapIterable()` as shown above seems to be a solution to the problem, although I'm not sure if it is the correct way to fix it.

## Your Environment

* Reactor version(s) used: reactor-kafka 1.3.11, reactor 3.4.17
* JVM version (`java -version`): 17.0.2
* OS and version (eg `uname -a`): Ubuntu 22.04

**Comment by rgrebski** [comment]
This issue is going to be fixed within #313.
PR created, waiting for review

**Comment by garyrussell** [comment]
Closed by https://github.com/reactor/reactor-kafka/commit/8f23e7773b986ed2a3ac08f9b412fbfc47b172d4
